BUILDING.md

```diff
@@ -31,4 +39,78 @@ The following trivial build script changes were made to successfully run `./grad
+
+## Developing new Plugins for OpenSearch
+
+All OpenSearch Plugins must contain the `plugin-descriptor.properties` file. The fields contained in the `plugin-descriptor.properties` file are mentioned [here](https://github.com/opensearch-project/OpenSearch/blob/main/buildSrc/src/main/resources/plugin-descriptor.properties). If using the gradle build system, these fields can be specified in the `build.gradle` file and the build system will create the properties file for us.
+
+- [](#)
+  - [Building Plugins with OpenSearch](#building-plugins-with-opensearch)
+    - [Publish OpenSearch to Maven Local](#publish-opensearch-to-maven-local)
+    - [Use OpenSearch from Maven Local in Plugins](#use-opensearch-from-maven-local-in-plugins)
+  - [Developing new Plugins for OpenSearch](#developing-new-plugins-for-opensearch)
+    - [Example `plugin-descriptor.properties` file](#example-plugin-descriptorproperties-file)
+    - [Example `build.gradle` file](#example-buildgradle-file)
+    - [Custom folder name for Plugins](#custom-folder-name-for-plugins)
+    - [Notes](#notes)
+
+### Example `plugin-descriptor.properties` file
+
+Following is a list of required and optional elements for the Plugins:
+
+```
+### mandatory elements for all plugins:
+#
+# 'description': simple summary of the plugin
+description=${description}
+#
+# 'version': plugin's version
+version=${version}
+#
+# 'name': the plugin name
+name=${name}
+#
+# 'classname': the name of the class to load, fully-qualified.
+classname=${classname}
+#
+# 'java.version': version of java the code is built against
+# use the system property java.specification.version
+# version string must be a sequence of nonnegative decimal integers
+# separated by "."'s and may have leading zeros
+java.version=${javaVersion}
+#
+# 'opensearch.version': version of opensearch compiled against
+opensearch.version=${opensearchVersion}
+### optional elements for plugins:
+#
+# 'custom.foldername': the custom name of the folder in which the plugin is installed.
+custom.foldername=${customFolderName}
+#
+#  'extended.plugins': other plugins this plugin extends through SPI
+extended.plugins=${extendedPlugins}
+#
+# 'has.native.controller': whether or not the plugin has a native controller
+has.native.controller=${hasNativeController}
+```
+
+### Example `build.gradle` file
+
+If adding the properties in `build.gradle` file, below is an example for the `opensearchplugin`:
+
+```
+opensearchplugin {
+    name 'opensearch-plugin'
+    description 'Example OpenSearch plugin'
+    classname 'org.opensearch.example.Plugin'
+}
+```
+### Custom folder name for Plugins
+
+A folder name property `customFolderName` can be specified for Plugins (for OpenSearch versions higher than 1.0.0) to give a custom folder name during plugin installation. Related PR in [OpenSearch](https://github.com/opensearch-project/OpenSearch): [PR #848](https://github.com/opensearch-project/OpenSearch/pull/848)
```

Comment:
There's no OpenSearch version older than 1.0, so no need to specify.

Say "See [opensearch#848](...) for more information."

Reply:
Sorry I meant to say that custom folder name property will be available for OpenSearch versions greater than 1.0.0.